charts/kube-prometheus-stack/templates/prometheus/additionalPrometheusRules.yaml

```diff
@@ -2,15 +2,15 @@
 apiVersion: v1
 kind: List
 metadata:
-  name: {{ include "kube-prometheus-stack.fullname" $ }}-additional-prometheus-rules
+  name: {{ printf "%s-%s" (include "kube-prometheus-stack.fullname" $) "additional-prometheus-rules" | trunc 63 | trimSuffix "-" }}
   namespace: {{ template "kube-prometheus-stack.namespace" . }}
 items:
 {{- if .Values.additionalPrometheusRulesMap }}
 {{- range $prometheusRuleName, $prometheusRule := .Values.additionalPrometheusRulesMap }}
   - apiVersion: monitoring.coreos.com/v1
     kind: PrometheusRule
     metadata:
-      name: {{ template "kube-prometheus-stack.name" $ }}-{{ $prometheusRuleName }}
+      name: {{ printf "%s-%s" (include "kube-prometheus-stack.fullname" $) $prometheusRuleName | trunc 63 | trimSuffix "-" }}
```

Comment:
What did you think about if `$prometheusRule.name` is set, the property will be preferred over the default generated one?

If someone does not like the new name, the has at least the chance to mimic the old behavior.

Reply:
Well...but it's a mandatory field.  It's the `key` in the `key:value` dictionary.  At that point may as well remove `include "kube-prometheus-stack.fullname" $` right?  unless i'm missing something?